umbraco/Umbraco-CMS
# Fixes #14565 - Empty DocType folders hidden

### Prerequisites


If there's an existing issue for this PR then this fixes [Issue #14565](https://github.com/umbraco/Umbraco-CMS/issues/14565)

### Description
~~I have added an additional method on the contentTypeService combines the check for any contentTypes with a check for any entityContainers. This is probably not the most performant way of doing this as this results in two checks. But during my quick overview I could not find any existing check for this. Please feel free to suggest an edit if I'm mistaken here..~~


ContentTypeTreeController now also checks for the existence of any folders when setting the hasChildren property on the tree. Previously this would only check for any contentTypes. 


~~I've added tests to the contentTypeService tests-class which pass on my local machine.~~

To verify fix, 

1. Fresh install of umbraco. 
2. Under settings, 'Document Types' should have no arrow, because it's empty. 
3. Create a folder under Document Types, the folder should be visible and 'Document Types' should have an expand-arrow. 
4. Reload the browser window. 'Documen Types' should have an arrow and should have the created folder as a descendant. 

Reproduction steps for this bug can be found in the original issue.
